### PR TITLE
Remove mcp specific test

### PR DESCRIPTION
**Pull request by ayj**
Depends on https://github.com/istio/istio/pull/10222 being merged first.

### CODE REVIEW

**Comment by ayj** [comment]
@hklai, non-MCP tests have been removed from presubmit

File: prow/config.yaml [hunk 1]
```diff
@@ -97,20 +97,26 @@ branch-protection:
...
               required_status_checks:
                 contexts:
                 - "ci/circleci: shellcheck"
+                - "ci/circleci: e2e-galley"
                 - "ci/circleci: e2e-pilot-auth-v1alpha3-v2"
+                - "ci/circleci: e2e-pilot-auth-v1alpha3-v2-non-mcp"
                 - "ci/circleci: e2e-pilot-noauth-v1alpha3-v2"
                 - "ci/circleci: e2e-mixer-noauth-v1alpha3-v2"
                 - "ci/circleci: e2e-pilot-cloudfoundry-v1alpha3-v2"
+                - "ci/circleci: e2e-mixer-noauth-v1alpha3-v2-non-mcp"
```
Comment (hklai) [inline prow/config.yaml]: is https://github.com/istio/istio/pull/10222 in master already?

Comment (ayj) [reply prow/config.yaml]: Not yet (see https://github.com/istio/istio/pull/10256).

**Review by hklai — APPROVED** [review]
/lgtm

**Pull request merged**